activerecord/test/models/destroy_async_parent.rb

```diff
@@ -4,12 +4,12 @@ class DestroyAsyncParent < ActiveRecord::Base
    self.primary_key = "parent_id"
 
    has_one :dl_keyed_has_one, dependent: :destroy_async,
-     foreign_key: :destroy_async_parent_id, primary_key: :parent_id
+     foreign_key: :destroy_async_parent_id
    has_many :dl_keyed_has_many, dependent: :destroy_async,
-     foreign_key: :many_key, primary_key: :parent_id
+     foreign_key: :many_key
    has_many :dl_keyed_join, dependent: :destroy_async,
-     foreign_key: :destroy_async_parent_id, primary_key: :joins_key
+     foreign_key: :destroy_async_parent_id
    has_many :dl_keyed_has_many_through,
      through: :dl_keyed_join, dependent: :destroy_async,
-     foreign_key: :dl_has_many_through_key_id, primary_key: :through_key
+     foreign_key: :dl_has_many_through_key_id
```

Comment:
I don't think this is the correct fix for these tests. They should pass without being changed so the issue seems to be in the code changes made.